File: lesson_4/hw_1.py
#Задание №1 Реализовать структуру данных стэк, который за О(1) выдает минимум в стэке.

class MinStack:
    stack = []

    def push(self, value: int):
        self.value = value
        self.stack.append(self.value)

    def pop(self):
        self.stack.pop()

    def top(self) -> int:
        return max(self.stack)

    def getMin(self) -> int:
        return min(self.stack)

minstack = MinStack()

minstack.push(1)
minstack.push(0)
minstack.push(-1)
minstack.push(-4)
minstack.top()
minstack.getMin()
minstack.pop()
minstack.getMin()

File: JS/js_lesson_3/hw_1.js
// Задание №1 С помощью цикла while вывести все простые числа в промежутке от 0 до 100.

let max = 100;
let n = 2;
let primeArr = [];

while (n <= max) {
    if (n == 2 || n == 3 || n == 5 || n == 7) {
        primeArr.push('' + n);
        n++;
        continue;
    } else if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0 || n % 7 == 0) {
        n++;
        continue;
    }
    primeArr.push('' + n++);
} 

console.log(primeArr)